HelpSource/Reference/Conditional-Execution.schelp

```
@@ -108,10 +140,41 @@ while { testFuncBody } { loopFuncBody };
 while ({ testFuncBody }) { loopFuncBody };
 // in receiver notation
 { testFuncBody }.while({ loopFuncBody })
-// distinct from C-style, keyword-based while, which will not work in SuperCollider:
+// distinct from C-style while, which will not work in SuperCollider:
 while (testFuncBody) { loopFuncBody }; // -> Syntax Error
 ::
 
+subsubsection:: Return values (.if, .case, .switch)
+
+Unlike other common programming languages, conditional expressions (code::.if::, code::.case::, and code::.switch::) in SuperCollider have return values by default;
```

This could be expressed more generally, in SuperCollider everything is an expression, i.e. returns a value. There are no pure statements as far as I know. Even `var x;` is an expression that returns `nil`.

So "have return values by default;" is a little misleading, because they always have a return value.

---

I agree with what is said, but I do think that `var x = 2;` and `var y;` are statements. It is just that it is impossible to write them in a situation where you'd depend on their return value so they appear like they return nil. Consider these three examples.

```
(
{
	var x;
	var y = 3;
	2
}.def.dumpByteCodes
)
BYTECODES: (2)
  0   65       PushSpecialValue 2
  1   F2       BlockReturn

(
{
	var x;
	var y = 3;
}.def.dumpByteCodes
)
BYTECODES: (2)
  0   6E       PushSpecialValue nil
  1   F2       BlockReturn
```
No bytecodes are emitted.

```
(
{
	var x = 2;
	var y = 3;
	x + y
}.def.dumpByteCodes
)

BYTECODES: (5)
  0   30		 PushTempZeroVar 'x'
  1   31		 PushTempZeroVar 'y'
  2   B0       TailCallReturnFromFunction
  3   E0       SendSpecialBinaryArithMsg '+'
  4   F2       BlockReturn


```
Only where they are used is the corresponding byte codes emitted. 

The bytecode `PushTempZeroVar` contains a 'drop' instruction. This means whatever is put on the stack is immediately removed from it, so it is like the return value doesn't exist, which I believe would be the lhs of `var x = n` or `nil` if no default is provided — `PushTempZeroVar` copies the default from the frame.